flask_log_request_id/request_id.py

```diff
@@ -81,7 +81,7 @@ def _persist_request_id():
                     setattr(g, g_object_attr, self._request_id_generator())
 
         # Register after request
```

Comment:
As a general pattern we should remove the `self.app = app` in the `__init__`, because this is not respecting the factory pattern either

Reply:
Yes I agree that it was wrong to keep work on the `self.app` object inside the `init_app` still it is not against holding the `app` inside the constructor. Through the documentation of flask:

> 
> Note on init_app
> As you noticed, init_app does not assign app to self. This is intentional! Class based Flask  extensions must only store the application on the object when the application was passed to the constructor. This tells the extension: I am not interested in using multiple applications.
> 
> When the extension needs to find the current application and it does not have a reference to it, it must either use the current_app context local or change the API in a way that you can pass the application explicitly.

https://flask.palletsprojects.com/en/1.1.x/extensiondev/